src/internal.c

```diff
@@ -10446,7 +10449,7 @@ static int GetRecordHeader(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         default:
 #ifdef OPENSSL_ALL
             {
-                char *method = (char*)input + start;
+                char *method = (char*)ssl->buffers.inputBuffer.buffer + start;
```

Comment:
I think that `GetInputData` calls inside `DtlsGetHeader` may trash the start of the buffer and invalidate `start`. I would consider adding a check not supporting plain HTTP requests in DTLS . Like adding `if (!options.dtls)`

Reply:
This block is now wrapped in `if (!ssl->options.dtls) { ... }`